components/dashboard/src/projects/RemoveProjectModal.tsx

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { FunctionComponent, useCallback, useState } from "react";
+import type { Project } from "@gitpod/gitpod-protocol";
+import { projectsService } from "../service/public-api";
+import { getGitpodService } from "../service/service";
+import ConfirmationModal from "../components/ConfirmationModal";
+import { useFeatureFlags } from "../contexts/FeatureFlagContext";
+
+type RemoveProjectModalProps = {
+    project: Project;
+    onClose: () => void;
+    onRemoved: () => void;
+};
+
+export const RemoveProjectModal: FunctionComponent<RemoveProjectModalProps> = ({ project, onClose, onRemoved }) => {
+    const { usePublicApiProjectsService } = useFeatureFlags();
+    const [disabled, setDisabled] = useState(false);
+
+    const removeProject = useCallback(async () => {
+        setDisabled(true);
+        usePublicApiProjectsService
+            ? await projectsService.deleteProject({ projectId: project.id })
+            : await getGitpodService().server.deleteProject(project.id);
+        setDisabled(false);
+        onRemoved();
+    }, [onRemoved, project.id, usePublicApiProjectsService]);
+
+    return (
+        <ConfirmationModal
+            title="Remove Project"
+            areYouSureText="Are you sure you want to remove this project from this team? Team members will also lose access to this project."
```

Comment:
**issue(non-blocking)**: This is out of the scope here, but there's a top margin here that's not needed in the modal component. Posting below how the modal would look like be ideally for visibility. However, feel free to leave this as is.

| BEFORE | AFTER |
|-|-|
| <img width="426" alt="modal-before" src="https://user-images.githubusercontent.com/120486/207282572-410ed44f-1688-4d3c-966e-cbc785497856.png"> | <img width="426" alt="modal-after" src="https://user-images.githubusercontent.com/120486/207282583-77e7565b-d6d4-4221-89af-9e2f55ac8b02.png"> |

Reply:
Got it, I'll update the confirmation modal in the follow up too.